### ISSUE TITLE
[2.0] [Linux] Client disconnects after suspend and there is no reconnect button

### ISSUE THREAD

**Issue by llucax** [owncloud/client#3888]
### Expected behaviour

After suspending the computer for a long time, Owncloud clients reconnects after connections is lost. Or at least there is a way to tell the program to reconnect an account
### Actual behaviour

A message saying "No connection to Owncloud URL. Unable to connect to URL/status.php timeout". It never tries to reconnect again, there is no apparent way to manually trigger a reconnection (that is not restarting the client) either.
### Steps to reproduce
1. Open owncloud
2. Suspend the computer
3. Owncloud doesn't reconnect properly
### Server configuration

Operating system: Ubuntu 15.04

Web server: Apache 2.4.10-9ubuntu1.1

Database: MySQL 5.6.25-0ubuntu0.15.04.1

PHP version: 5.6.4+dfsg-4ubuntu6.2

ownCloud version: 8.1.3-13.1

Storage backend: ?
### Client configuration

Client version: 2.0.0+dfsg-1

Operating system: Ubuntu 15.04

OS language: EN

Installation path of client: debian package
### Logs

TODO: I restarted with `--logwindow`, I will report back when I can reproduce it.
1. Output of `owncloud --logwindow` or `owncloud --logfile log.txt`
   (On Windows using `cmd.exe`, you might need to first `cd` into the ownCloud directory)
   (See also http://doc.owncloud.org/desktop/1.8/troubleshooting.html#client-logfile )
2. Web server error log:

No applicable, I guess. Going to the owncloud web interface all works fine.
1. ownCloud log (data/owncloud.log):

No applicable, I guess. Going to the owncloud web interface all works fine.

**Comment by tflidd** [comment]
I can confirm this. I just had the problem yesterday on a public wifi where you need to authenticate first. The client remains in a "no connection"-status (I waited for 2 minutes). Before (<2.0.0) it helped sometimes to pause the sync and re-enable it. Now you need to close and restart the client.
Client 2.0.1 (current stable).
OS X 10.9

**Comment by guruz** [comment]
@llucax Which Qt version do you see in the settings dialog?

**Comment by llucax** [comment]
I can't see the QT version anywhere in the settings dialog, but...
```
luca@nibbler:~ ✔ $ ldd /usr/bin/owncloud  | grep -i qt.*core
	libQt5Core.so.5 => /usr/lib/x86_64-linux-gnu/libQt5Core.so.5 (0x00007fda82157000)
luca@nibbler:~ ✔ $ dpkg -S /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
libqt5core5a:amd64: /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
luca@nibbler:~ ✔ $ apt list libqt5core5a
Listing... Done
libqt5core5a/testing,unstable,now 5.4.2+dfsg-9 amd64 [installed,automatic]
```

So I would say something like **5.4.2+dfsg-9**.

I waited much more than 2 minutes, and it never seems to reconnect. At one point by poking with stuff in the configuration I managed to trigger a reconnect, after asking for the password again.

I too, with the old version < 2.0 enabled and disabled syncing and it reconnected, but it also happened extremely rarely, now I get into this disconnected state much more often (although, not every time I suspend the computer).

In any case, having a connect button enabled when the client is disconnected seems like a reasonable idea...

**Comment by mailme-gx** [comment]
I can also confirm this happens since the last upgrade need to quit restart the client to connect, used to work fine. Using Arch linux with package versions:
owncloud-client: 2.0.2-1
qtkeychain-qt5: 0.5-1
qt5-webkit: 5.5.1-2
hicolor-icon-theme: 0.15-1
xdg-utils: 1.1.1-1

**Comment by guruz** [comment]
Wondering if related to https://github.com/owncloud/client/issues/4051 or https://github.com/owncloud/client/issues/3600

**Comment by ThomasLeister** [comment]
Same here on my two Arch Linux machines with OwnCloud Client 2.0.2 (Qt Version: 5.5.1). (One of them connected via WLAN, the other one connected via LAN). After resuming from standby the OwnCloud Client can't start a network connection. It says: "Access to network denied".

Messages from the client log:

```
10-25 12:39:12:383 7117 OCC::ConnectionValidator::checkAuthentication: # Check whether authenticated propfind works.
10-25 12:39:12:384 7117 OCC::AbstractNetworkJob::start: !!! OCC::PropfindJob created for "https://cloud.myserver.net" + "/"
10-25 12:39:12:385 7117 OCC::AbstractNetworkJob::slotFinished: void OCC::AbstractNetworkJob::slotFinished() QNetworkReply::NetworkError(UnknownNetworkError) "Der Zugriff auf das Netzwerk ist nicht gestattet." QVariant(Invalid)
10-25 12:39:12:385 7117 unknown: QIODevice::read (QDisabledNetworkReply): device not open
10-25 12:39:12:386 7117 OCC::AccountState::slotConnectionValidatorResult: AccountState connection status change:  "Connected" -> "Timeout"
10-25 12:39:12:386 7117 OCC::AccountState::setState: AccountState state change:  "Verbunden" -> "Netzwerkfehler"
10-25 12:39:12:386 7117 OCC::AccountSettings::showConnectionLabel: "Keine Verbindung zu ownCloud auf <a href=\"https://cloud.myserver.net\">https://cloud.myserver.net</a>\nDer Zugriff auf das Netzwerk ist nicht gestattet."
10-25 12:39:12:387 7117 OCC::FolderMan::slotAccountStateChanged: Account "user@cloud.myserver.net" disconnected, terminating or descheduling sync folders
10-25 12:39:12:388 7117 OCC::PropfindJob::finished: PROPFIND request *not* successful, http result code is 0 ""
10-25 12:39:45:289 7117 OCC::ConnectionValidator::checkServerAndAuth: Checking server and authentication
10-25 12:39:45:290 7117 OCC::ConnectionValidator::checkServerAndAuth: Trying to look up system proxy
10-25 12:39:45:291 7439 OCC::SystemProxyRunnable::run: virtual void OCC::SystemProxyRunnable::run() Starting system proxy lookup
10-25 12:39:45:293 7117 OCC::ConnectionValidator::systemProxyLookupDone: void OCC::ConnectionValidator::systemProxyLookupDone(const QNetworkProxy&) Setting QNAM proxy to be system proxy "2://:0"
10-25 12:39:45:293 7117 OCC::AbstractNetworkJob::start: !!! OCC::CheckServerJob created for "https://cloud.myserver.net" + "status.php"
10-25 12:39:45:294 7117 OCC::AbstractNetworkJob::slotFinished: void OCC::AbstractNetworkJob::slotFinished() QNetworkReply::NetworkError(UnknownNetworkError) "Der Zugriff auf das Netzwerk ist nicht gestattet." QVariant(Invalid)
10-25 12:39:45:295 7117 OCC::ConnectionValidator::slotNoStatusFound: void OCC::ConnectionValidator::slotNoStatusFound(QNetworkReply*) QNetworkReply::NetworkError(UnknownNetworkError) "Der Zugriff auf das Netzwerk ist nicht gestattet."
10-25 12:39:45:295 7117 OCC::AccountState::slotConnectionValidatorResult: AccountState connection status change:  "Timeout" -> "Status not found"
10-25 12:39:45:296 7117 OCC::AccountSettings::showConnectionLabel: "Keine Verbindung zu ownCloud auf <a href=\"https://cloud.myserver.net\">https://cloud.myserver.net</a>\nDer Zugriff auf das Netzwerk ist nicht gestattet."
10-25 12:39:45:297 7117 OCC::FolderMan::slotAccountStateChanged: Account "user@cloud.myserver.net" disconnected, terminating or descheduling sync folders
10-25 12:39:45:297 7117 unknown: QIODevice::read (QDisabledNetworkReply): device not open
10-25 12:39:45:297 7117 OCC::CheckServerJob::finished: error: status.php replied  0 ""
10-25 12:40:17:286 7117 OCC::ConnectionValidator::checkServerAndAuth: Checking server and authentication
10-25 12:40:17:286 7117 OCC::ConnectionValidator::checkServerAndAuth: Trying to look up system proxy
10-25 12:40:17:287 7446 OCC::SystemProxyRunnable::run: virtual void OCC::SystemProxyRunnable::run() Starting system proxy lookup
10-25 12:40:17:287 7117 OCC::ConnectionValidator::systemProxyLookupDone: void OCC::ConnectionValidator::systemProxyLookupDone(const QNetworkProxy&) Setting QNAM proxy to be system proxy "2://:0"
10-25 12:40:17:288 7117 OCC::AbstractNetworkJob::start: !!! OCC::CheckServerJob created for "https://cloud.myserver.net" + "status.php"
10-25 12:40:17:288 7117 OCC::AbstractNetworkJob::slotFinished: void OCC::AbstractNetworkJob::slotFinished() QNetworkReply::NetworkError(UnknownNetworkError) "Der Zugriff auf das Netzwerk ist nicht gestattet." QVariant(Invalid)
10-25 12:40:17:289 7117 OCC::ConnectionValidator::slotNoStatusFound: void OCC::ConnectionValidator::slotNoStatusFound(QNetworkReply*) QNetworkReply::NetworkError(UnknownNetworkError) "Der Zugriff auf das Netzwerk ist nicht gestattet."
10-25 12:40:17:289 7117 OCC::AccountSettings::showConnectionLabel: "Keine Verbindung zu ownCloud auf <a href=\"https://cloud.myserver.net\">https://cloud.myserver.net</a>\nDer Zugriff auf das Netzwerk ist nicht gestattet."
10-25 12:40:17:290 7117 OCC::FolderMan::slotAccountStateChanged: Account "user@cloud.myserver.net" disconnected, terminating or descheduling sync folders
10-25 12:40:17:290 7117 unknown: QIODevice::read (QDisabledNetworkReply): device not open
10-25 12:40:17:291 7117 OCC::CheckServerJob::finished: error: status.php replied  0 ""
```

**Comment by guruz** [comment]
@ThomasLeister @mailme-gx Since you guys use ArchLinux.. Do you have the chance to build your Qt with https://codereview.qt-project.org/#/c/121724/ and see if that fixes it? 
This is not in Qt 5.5.1 but in the Qt 5.6 branch.

**Comment by guruz** [comment]
Patch also here: https://gist.github.com/guruz/47d5e9d47c76f1e6d371

**Comment by ThomasLeister** [comment]
Yes, I will try.

**Comment by ThomasLeister** [comment]
It perfectly works with the patch :)

**Comment by hammerheadlemon** [comment]
Having the same problem with Debian package version (Version 2.0.2+dfsg-1+b1). But reported there at https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=804883.

owncloud client does not resume network connection after laptop suspends.

Tail of log:
```
11-13 05:30:17:535 28130 OCC::AbstractNetworkJob::slotFinished: void OCC::AbstractNetworkJob::slotFinished() QNetworkReply::NetworkError(UnknownNetworkError) "Network access is disabled." QVariant(Invalid)
11-13 05:30:17:535 28130 OCC::ConnectionValidator::slotNoStatusFound: void OCC::ConnectionValidator::slotNoStatusFound(QNetworkReply*) QNetworkReply::NetworkError(UnknownNetworkError) "Network access is disabled."
11-13 05:30:17:535 28130 OCC::AccountState::slotConnectionValidatorResult: AccountState connection status change:  "Timeout" -> "Status not found"
11-13 05:30:17:535 28130 OCC::AccountSettings::showConnectionLabel: "No connection to ownCloud at <a href=\"https://removedurl\">removedurl</a>.\nNetwork access is disabled."
11-13 05:30:17:535 28130 OCC::FolderMan::slotAccountStateChanged: Account "removedaccount" disconnected, terminating or descheduling sync folders
11-13 05:30:17:535 28130 unknown: QIODevice::read (QDisabledNetworkReply): device not open
11-13 05:30:17:535 28130 OCC::CheckServerJob::finished: error: status.php replied  0 ""
``` 
From apt: QT: 5.5.1+dfsg-6 

How do you apply the patch?

Thanks.

**Comment by ThomasLeister** [comment]
@hammerheadlemon you can download the patched version via:
```
mkdir myqt
cd myqt
git init
git fetch https://codereview.qt-project.org/qt/qtbase refs/changes/24/121724/14 && git checkout FETCH_HEAD
```

and follow the instructions at https://wiki.qt.io/Building_Qt_5_from_Git

I added --prefix /usr/local to my ./configure command:

```
./configure --prefix=/usr/local 
```

Don't forget to uninstall your QT5 before installing your self-compiled version via "make install". Please note that a simple "make install" will install qt directly into your filesystem. Your package manager (apt) won't recognize your new QT version. A better solution is to create a apt package from the compiled source and let apt install the new QT version... but I don't know how to do that.

**Comment by hammerheadlemon** [comment]
@ThomasLeister Thank you. I'll give it a go.

**Comment by mnlipp** [comment]
I can confirm the problem for version 2.0.2 (build from https://github.com/owncloud/client/commit/78d782601e4afb9130cd8b4e214b56bbe0d1c39a) using Qt 5.4.1 on Fedora. It is extremely annoying since I use owncloud to synchronize files between laptops that I suspend/resume several times a day. Of course, I often forget to restart the owncloud client after resume, so effectively owncloud simply doesn't work for me currently. I'd consider this a high priority issue. Why isn't it possible to issue a 2.0.3 version if the fix is available?

**Comment by gruentee** [comment]
I can confirm the issue on ArchLinux, too.

**Comment by ckamm** [comment]
@mnlipp The fix is in for a upstream library (Qt). Releasing a new client version wouldn't help since we don't know a workaround.

**Comment by danimo** [comment]
@mnlipp You can ask fedora if they would backport the patch.

**Comment by mnlipp** [comment]
Well, thanks, it works again with fedora since the upgrade to owncloud-client-2.1.0. I don't really know if they also updated the Qt libs (it's 5.5.1-8 currently), I didn't track them. Anyway, with an up-to-date fc22 the client reconnects again after the suspend.

**Comment by AsavarTzeth** [comment]
I too got the 2.1.0 update recently, on ArchLinux. A week ago to be precise.

The issue seems to be resolved with the update. I am also relatively confident  that the issue is fixed in the owncloud-client and not the Qt libs.

To test this I downgraded only the owncloud-client package to 2.0.2 and suspended my laptop. The issue reappeared. One package upgrade later and the issue is fixed.

**Comment by guruz** [comment]
Please also see discussion in https://github.com/owncloud/client/issues/4051

**Comment by guruz** [comment]
(For OS X and Windows, we add the patch https://github.com/owncloud/client/blob/master/admin/qt/patches/0012-Make-sure-networkAccessibilityChanged-is-emitted.patch which did not make it into Qt 5.1.1 release)

**Comment by Paviluf** [comment]
What about Linux ?

**Comment by de-live-gdev** [comment]
I am on Ubuntu 16.04, v2.2.0 (build 1605) and this still happens

**Comment by danimo** [comment]
@de-live-gdev This is an upstream bug with Ubuntu, which needs reporting. I'll get to this next week unless someone beats me to it.

**Comment by de-live-gdev** [comment]
Ah okay, just a sidenote: The build comes from the owncloud ppa, not from Ubuntu repos

**Comment by danimo** [comment]
@de-live-gdev Yes, but it uses Qt from Ubuntu, which has the bug.

**Comment by tjyrinki** [comment]
Note that Ubuntu 16.04 LTS already includes https://codereview.qt-project.org/#/c/121724/ (as does Debian) so that alone is not seemingly enough.

**Comment by guruz** [comment]
@tjyrinki Maybe another patch is needed too: https://github.com/owncloud/client/issues/4720#issuecomment-221543854

**Comment by tjyrinki** [comment]
That's included too, the patch is by an Ubuntu developer originally to fix an issue on Ubuntu Phone so it has been in Ubuntu (also 16.04 LTS) for some time already.

**Comment by llucax** [comment]
> That's included too, the patch is by an Ubuntu developer originally to
> fix an issue on Ubuntu Phone so it has been in Ubuntu (also 16.04 LTS)
> for some time already.

For what is worth, I'm using both Ubuntu 14.04 (trusty) and Debian
testing and I didn't have this issue anymore or a long time now.

**Comment by guruz** [comment]
We pushed a possible workaround. Could you please test 2.2.1 and report back if it works? Thanks

**Comment by cperreault** [comment]
Version 2.2.1* works fine for me now on Ubuntu 16.04! The synchronization restarts automatically when the network connection resumes. Thanks.

*Built from Git revision 1a6283 on Jun 7 2016, 15:31:22 using Qt 5.5.1, OpenSSL 1.0.2g-fips 1 Mar 2016

**Comment by caco3** [comment]
Seems to be fixed on 2.2.1 using Kubuntu 16.04, KDE Neon
Thank you very much!

**Comment by mortie** [comment]
I'm running Arch Linux, which uses Qt 5.8.0-2, and the issue still occurs. After suspend, the client sometimes doesn't reconnect, just displaying a "Network access is disabled" error. The only reliable way I have found to fix that is to restart NetworkManager; pausing and unpausing syncing does nothing.

**Comment by timorei** [comment]
Issue happens on KDE Neon User
plasma-shell 5.9.2
owncloud-client 2.1.1
Qt 5.7.1

**Comment by guruz** [comment]
@timorei The current client version is 2.2.4 not 2.2.1.
Also we brought out 2.3.0beta1 which you could test :-)
What is the exact error message you get?

@mortie Thanks, @ogoffart is having a look.

**Comment by ogoffart** [comment]
Seems like another issue that should be fixed with https://github.com/owncloud/client/pull/5538

**Comment by ogoffart** [comment]
Should be fixed in 2.3.0 beta2

**Comment by mabkenar** [comment]
I never noticed this problem in any client version I had, whether in Ubuntu 14.04 or 16.04 with packages from openSUSE build service. 

Right now, I compiled version 2.3.0git of the client on KDE Neon, and it is also working fine.
